mongo_connector/dest_mapping.py

```diff
@@ -15,143 +15,230 @@
+        self.exclude_fields = exclude_fields
+
         # the input namespace_set and ex_namespace_set could contain wildcard
-        self.namespace_set = namespace_set
-        self.ex_namespace_set = ex_namespace_set
+        self.namespace_set = set()
+        self.ex_namespace_set = RegexSet.from_namespaces(
+            ex_namespace_set or [])
 
         self.lock = threading.Lock()
 
+        user_mapping = user_mapping or {}
+        namespace_set = namespace_set or []
         # initialize
         for ns in namespace_set:
             user_mapping.setdefault(ns, ns)
 
-        for k, v in user_mapping.items():
-            self.set(k, v)
-
-    def set_plain(self, key, value):
-        """A utility function to set the corresponding plain variables"""
-        if value in self.reverse_plain:
-            raise errors.InvalidConfiguration(
-                "Destination namespaces set should not"
-                " contain any duplicates.")
-
-        db, col = key.split(".", 1)
-        self.plain[key] = value
-        self.reverse_plain[value] = key
-        if col != "$cmd":
-            if db not in self.plain_db:
-                self.plain_db[db] = set([value.split(".")[0]])
+        for src_name, v in user_mapping.items():
+            if isinstance(v, dict):
+                self._add_mapping(src_name, v.get('rename'))
             else:
-                self.plain_db[db].add(value.split(".")[0])
-
-    def match(self, src, dst_pattern):
-        """If source string src matches dst, return the matchobject"""
-        reg_pattern = r'\A' + dst_pattern.replace('*', '(.*)') + r'\Z'
-        m = re.match(reg_pattern, src)
-        return m
+                self._add_mapping(src_name, v)
 
-    def match_set(self, plain_src, dst_arr):
-        for x in dst_arr:
-            if plain_src == x:
-                return True
-            if "*" in x:
-                m = self.match(plain_src, x)
-                if m:
-                    return True
+    def _add_mapping(self, src_name, dest_name=None, include_fields=None,
+                     exclude_fields=None):
+        if (self.include_fields and exclude_fields or
+                self.exclude_fields and include_fields or
+                include_fields and exclude_fields):
+            raise errors.InvalidConfiguration(
+                "Cannot mix include fields and exclude fields in "
+                "namespace mapping for: '%s'" % (src_name,))
+        if dest_name is None:
+            dest_name = src_name
+        self.set(MappedNamespace(dest_name=dest_name, source_name=src_name,
+                                 include_fields=include_fields,
+                                 exclude_fields=exclude_fields))
+        # Add the namespace for commands on this database
+        cmd_name = src_name.split('.', 1)[0] + '.$cmd'
+        dest_cmd_name = dest_name.split('.', 1)[0] + '.$cmd'
+        self.set(MappedNamespace(dest_name=dest_cmd_name, source_name=cmd_name))
```

Comment:
`set` is usually the name of a method that associates a value with a key; this looks like it's just adding something.